### PR TITLE
Fix gcrs posvel

### LINKED ISSUE #6680: EOP in Python (Precession Nutation)
Hi,
I am looking for a way to query the IERS bulletin and obtain the Earth Orientation Parameters in Python. The analogue of the IDL [hprstatn.pro](https://harrisgeospatial.com/docs/hprstatn.html) procedure.

Just checking if this is implemented in some form in Astropy. I know that I can query the IERS bulletin and obtain the table, but is the final product already done? 

My end goal is to obtain the x,y,z,vx,vy,vz of an observatory wrt the barycenter taking into consideration these movements.

### LINKED ISSUE #6697: ```EarthLocation.get_gcrs_posvel``` is innacurate
Bit of an embarrassing one this, but my own routine ```get_gcrs_posvel``` assumes the rotation axis of the earth is aligned with the GCRS pole. This introduces velocities errors on the order of 1 meter/sec

### PR DESCRIPTION
**Pull request by StuartLittlefair**
Fixes #6697 and also fixes #6680, subject to the precision required by that user.

I've removed the incorrect assumption that the rotation axis is aligned with the GCRS pole. This routine now calculates the rotation of the earth in the CIRS frame and rotates into GCRS. This could have been done using the new ability to include differentials in frames, but the current way seemed neater to me.

### CODE REVIEW

**Comment by StuartLittlefair** [comment]
Also fixes #6680, subject to the precision required by that user.

**Comment by bsipocz** [comment]
> Also fixes #6680, subject to the precision required by that user.

can you mention that in the PR description commit, so merging will autoclose that issue, too?

File: CHANGES.rst [hunk 1]
```diff
@@ -428,6 +428,8 @@ astropy.coordinates
 - Fixed a bug that was preventing ``SkyCoord`` objects made from lists of other
   coordinate objects from being written out to ECSV files. [#6448]
 
+- Improved accuracy of velocity calculation in ```EarthLocation.get_gcrs_posvel```. [#6697]
```
Comment (bsipocz) [inline CHANGES.rst]: release 2.0.2 is already out, please move the entry to 2.0.3

**Comment by MSeifert04** [comment]
I think astrobot doesn't recognize the PR number because it's off by 2.

**Comment by StuartLittlefair** [comment]
@eteq for comparison, this is what the routine would look like using differentials:

```python
    def get_gcrs_posvel(self, obstime):
        """
        Calculate the GCRS position and velocity of this object at the
        requested ``obstime``.

        Parameters
        ----------
        obstime : `~astropy.time.Time`
            The ``obstime`` to calculate the GCRS position/velocity at.

        Returns
        --------
        obsgeoloc : `~astropy.coordinates.CartesianRepresentation`
            The GCRS position of the object
        obsgeovel : `~astropy.coordinates.CartesianRepresentation`
            The GCRS velocity of the object
        """
        # do this here to prevent a series of complicated circular imports
        from .builtin_frames import CIRS
        from .builtin_frames.intermediate_rotation_transforms import gcrs_to_cirs_mat

        itrs = self.get_itrs(obstime)
        cirs_frame = CIRS(obstime=obstime)
        cirs_cart = itrs.transform_to(cirs_frame).cartesian
        rotation_vector = CartesianRepresentation(0/u.s, 0/u.s, OMEGA_EARTH)
        cirs_vel = rotation_vector.cross(cirs_cart)
        cirs_posvel=cirs_cart.with_differentials({'s': coord.CartesianDifferential(*cirs_vel.xyz)})
        gcrs_posvel = cirs_posvel.transform(matrix_transpose(gcrs_to_cirs_mat(t)))
        return gcrs_posvel.without_differentials(), gcrs_posvel.differentials['s'].to_cartesian()
```

Personally I find it a bit harder to follow, but it is marginally faster, as it has one fewer transformation...

**Comment by StuartLittlefair** [comment]
Documentation failures look unrelated to changes in this PR @bsipocz ?

**Comment by bsipocz** [comment]
Yes, those warnings are due to #6695

**Comment by mhvk** [comment]
This looks good to me. I actually like the use of the standard transformation routines, but think it could also be left to a separate issue.

Continuing separate though, might it be more logical to make a new routine `get_gcrs` that returns the full gcrs position with derivative, and is used here to split out the two parts. Or even to have a more general `.transform_to` method in analogy with `SkyCoord`?

**Comment by eteq** [comment]
(Just as a temporary note: this is on my radar and aim to review it ASAP)

**Comment by StuartLittlefair** [comment]
Superceded by #6861

**Pull request closed without merging**